server/routes/pdf.js

```diff
@@ -39,6 +39,115 @@ module.exports = ({ pdfService, prisonerService }) => (router, audited) => {
     res.redirect(`/hdc/pdf/taskList/${decision}/${bookingId}`)
   })
 
+  router.get(
+    '/selectLicenceType/:bookingId',
+    asyncMiddleware(async (req, res) => {
+      const { bookingId } = req.params
+
+      const prisoner = await prisonerService.getPrisonerPersonalDetails(bookingId, res.locals.token)
+      let errors = firstItem(req.flash('errors')) || {}
+
+      let licenceTemplateId = getIn(res.locals.licence, ['approvedVersionDetails', 'template'])
+      const offenceCommittedBeforeFeb2015 = getIn(res.locals.licence.licence, [
+        'document',
+        'template',
+        'offenceCommittedBeforeFeb2015',
+      ])
+
+      let offenceBeforeCutoff = ''
+
+      if (Object.keys(errors).includes('licenceTypeRadioListYes')) {
+        offenceBeforeCutoff = 'Yes'
+        const errorMessage = errors.licenceTypeRadioListYes
+        errors = { licenceTypeRadioList: errorMessage }
+        licenceTemplateId = ''
+      } else if (Object.keys(errors).includes('licenceTypeRadioListNo')) {
+        offenceBeforeCutoff = 'No'
+        const errorMessage = errors.licenceTypeRadioListNo
+        errors = { licenceTypeRadioList: errorMessage }
+        licenceTemplateId = ''
+      } else if (offenceCommittedBeforeFeb2015 === 'Yes' || offenceCommittedBeforeFeb2015 === 'No') {
+        offenceBeforeCutoff = offenceCommittedBeforeFeb2015
+      }
+
```

Comment:
Better to extract a function and use return values to initialise a const.

Reply:
Done